ecommerce/qverse_features/paystack/client.py

```diff
@@ -0,0 +1,128 @@
+import json
+import logging
+
+import requests
+
+from ecommerce.qverse_features.paystack import constants as paystack_const
+from ecommerce.qverse_features.paystack.exceptions import (
+    InvalidClientArgument,
+    InvalidPaystackClientMethod,
+    InvalidRequestMethod,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class PaystackClient(object):
+    """
+    Client for Paystack API requests
+    """
+
+    _CONTENT_TYPE = "application/json"
+
+    def __init__(self, base_url, authorization_key=None):
+        if base_url and authorization_key:
+            self._BASE_END_POINT = base_url
+            self._AUTHORIZATION_KEY = authorization_key
+        else:
+            msg = "Authorization key" if not authorization_key else "Base Url"
```

Comment:
This will raise only `Authorization key` error if 'base_url` and `authorization_key` both are absent